assembler/manual.html

```diff
@@ -848,12 +850,12 @@ <h4>Advanced options</h4>
 
 <p>
     <code>-t &lt;int></code> (or <code>--threads &lt;int></code>)<br>
-    &nbsp;&nbsp;&nbsp;&nbsp;Number of threads. The default value is 16.
+    &nbsp;&nbsp;&nbsp;&nbsp;Number of threads. The default value is 16. This is merely a maximum number of threads to be used. For example, with --threads 112 the initial the raw read error-correction step uses only 8 threads anyway.
```

Comment:
The "merely" is certainly misleading. And your example is misleading as well - certainly one cannot utilize more CPU threads than number of CPUs on your system. Hence the limitation of 8 on your case.

Reply:
No, the machine has 112 CPUs. Anyway, I inferred this from reading https://github.com/ablab/spades/issues/19#issuecomment-300307210 . And reading all opened Issues to learn what the tool really does is not ideal. Hence my attempt to come up with improvements.

Reply:
I hope you understand that in some cases there is no way to distribute the tasks evenly across the CPUs. And if, say, 104 chunks would require 1 second to run and another 8 chunks would require 1 minute to run, then processing them in parallel using 112 threads would look like as only 8 threads are being used. And answering to the original change - no, the read error correction step utilizies all the available CPUs. That comment in #19 does not bring anything new wrt the CPU usage.

Reply:
I already deleted the .cfg files in the output folder but there were often hardcoded values like 8 or 16 cpu for the tools. I suspect your template files do not get updated. I passed -t 112 to spades.py, it is its jobs to pass the information down to other tools.

In principle I agree that it is not always possible/wise to use all CPU cores but this is not my case.

Reply:
> I passed -t 112 to spades.py, it is its jobs to pass the information down to other tools.

It certainly does this.

> In principle I agree that it is not always possible/wise to use all CPU cores but this is not my case.

How have you found that only 8 CPU cores were always used?

Reply:
See `Splitting kmer instances into 128 buckets using 8 threads` in the output log and also see the CPU usage figure attached to https://github.com/ablab/spades/issues/65 .

Reply:
There might be multiple reasons why the number of CPU cores are lower than one requested. Sometimes on cluster systems one should explicitly specify the amount of CPU cores requested to job system and *forward* this value to OpenMP runtime. Unfortunately, all these details are heavily platform-specific and we cannot perform these steps for user.